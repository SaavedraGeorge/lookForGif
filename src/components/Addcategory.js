import { useState } from "react";
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    // { setCategories } es la destructuracion del props
    const [inputValue, setinputValue] = useState('');

    const handleInputChange = (e) => {
        setinputValue(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        //previene el comportamiento defoult de submit, no actualiza la pagina completa
        if (inputValue.trim().length > 2) {
            // trim() borra los espacios antes y despues
            setCategories(categorias => [inputValue, ...categorias]);
            setinputValue('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type='text'
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}