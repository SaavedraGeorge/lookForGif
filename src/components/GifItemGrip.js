import React from 'react';
import PropTypes from 'prop-types';

export const GifItemGrip = ({id, title, url}) => {
    return (
        <div className="card animate__animated animate__fadeIn">
            <p>{title}</p>
            <img src={url} alt={title} />
        </div>
    )
}
GifItemGrip.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}