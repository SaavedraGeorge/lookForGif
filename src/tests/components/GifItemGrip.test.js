import React from 'react';
import {shallow} from 'enzyme';

import {GifItemGrip} from '../../components/GifItemGrip';

describe('componente gif grip', () => {
    const title = 'titulo'
    const url = 'http://www.google.cl'
    const wrapper = shallow(<GifItemGrip url ={url} title={title}/>);

    test('mostrar <GifItemGrip /> ', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('should look class in the divs', () => {
        const div = wrapper.find('div');
        const nameClass = div.prop('className');
        const clases = 'card animate__animated animate__fadeIn'

        expect(nameClass).toBe(clases);
        expect(nameClass.includes('animate__fadeIn')).toBe(true);
    })
    
    
})
