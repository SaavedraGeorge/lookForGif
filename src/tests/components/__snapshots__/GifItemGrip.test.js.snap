// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`componente gif grip mostrar <GifItemGrip />  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <p>
    titulo
  </p>
  <img
    alt="titulo"
    src="http://www.google.cl"
  />
</div>
`;
